plugin-annotation/src/main/java/com/mapbox/maps/plugin/annotation/AnnotationManagerImpl.kt

```diff
@@ -94,6 +95,12 @@ abstract class AnnotationManagerImpl<G : Geometry, T : Annotation<G>, S : Annota
     gesturesPlugin.addOnMapClickListener(mapClickResolver)
     gesturesPlugin.addOnMapLongClickListener(mapLongClickResolver)
     gesturesPlugin.addOnMoveListener(mapMoveResolver)
+    delegateProvider.mapListenerDelegate.addOnDidFinishRenderingMapListener {
```

Comment:
@Chaoba Instead of OnDidFinishRenderingMapListener, we could use MapStyleObserverPlugin on the AnnotationPlugin, and propagate the style changed event to the underlying annotation managers to re-add the layers/sources.

Reply:
ticketed at https://github.com/mapbox/mapbox-maps-android/issues/146